File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if(word1.length() != word2.length()) return false; 

        vector<int> word1Map(26,0); 
        vector<int> word2Map(26,0); 

        for(int i = 0; i < word1.size(); i++){
            word1Map[word1[i] - 'a']++; 
            word2Map[word2[i] - 'a']++; 
        }

        for(int i = 0; i < word1Map.size(); i++){
            //if(word1Map[i] != word2Map[i]) return false; 
            if((word1Map[i] == 0 && word2Map[i] > 0) || (word2Map[i] == 0 && word1Map[i] > 0)){
                return false; 
            }
        } 

        sort(word1Map.begin(),word1Map.end()); 
        sort(word2Map.begin(),word2Map.end()); 


        return word1Map == word2Map; 
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if(word1.length() != word2.length()) return false;

        Map<Character,Integer> mapA = new HashMap<>();
        Map<Character,Integer> mapB = new HashMap<>();

        for(char c : word1.toCharArray()) mapA.merge(c,1,Integer::sum);
        for(char c : word2.toCharArray()){
            if(!mapA.containsKey(c)) return false;
            mapB.merge(c,1,Integer::sum);
        }

        // // 키 집합도 확인 (word1 글자가 word2에 있는지 - 양방향)
        // if(!mapA.keySet().equals(mapB.keySet())) return false;

        // Set → 정렬 List (개수 유지)
        List<Integer> listA = new ArrayList<>();
        List<Integer> listB = new ArrayList<>();

        for(var e : mapA.entrySet()) listA.add(e.getValue());
        for(var e : mapB.entrySet()) listB.add(e.getValue());

        Collections.sort(listA);
        Collections.sort(listB);

        return listA.equals(listB);
    }
}